network/validator/pubsub/authorized_sender_validator.go

```diff
@@ -8,38 +10,34 @@ import (
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"github.com/rs/zerolog"
 
-	channels "github.com/onflow/flow-go/engine"
 	"github.com/onflow/flow-go/network"
 
 	"github.com/onflow/flow-go/model/flow"
-	cborcodec "github.com/onflow/flow-go/network/codec/cbor"
-	"github.com/onflow/flow-go/network/message"
 )
 
 // AuthorizedSenderValidator using the getIdentity func will check if the role of the sender
 // is part of the authorized roles list for the channel being communicated on. A node is considered
-// to be authorized to send a message if all of the following are true.
-// 1. The node is authorized.
-// 2. The message type is a known message type (can be decoded with cbor codec).
-// 3. The authorized roles list for the channel contains the senders role.
-// 4. The node is not ejected
+// to be authorized to send a message if the following are true.
+// 1. The node is staked.
+// 2. The node is not ejected.
+// 3. The message type is a known message type (can be decoded with network codec).
+// 4. The message is authorized to be sent on channel.
+// 4. The sender role is authorized to send message channel.
+// 5. The sender role is authorized to participate on channel.
 func AuthorizedSenderValidator(log zerolog.Logger, channel network.Channel, getIdentity func(peer.ID) (*flow.Identity, bool)) MessageValidator {
```

Comment:
I think we should add direct unit test coverage for `AuthorizedSenderValidator`

Reply:
There is an issue to add test that cover each message code https://github.com/dapperlabs/flow-internal/issues/1718 , this is separate from the work done in this PR which is refactoring network codec etc but not changing logic in authorized sender validator.